Main.java

```diff
@@ -0,0 +1,88 @@
+public class Main {
+
+    private static final boolean left = true;
+    private static final boolean right = false;
+
+    public static void main(String[] args) {
+
+        // Test the tree from the assignment
+        BinaryTree<Integer> myTree = new BinaryTree(16);
+        myTree.insert(9, 16 , left);
+        myTree.insert(18, 16, right);
+        myTree.insert(3, 9, left);
+        myTree.insert(14, 9, right);
+        myTree.insert(1, 3, left);
+        myTree.insert(5, 3, right);
+        myTree.insert(19, 18, right);
+        printAncestors(myTree, 5);  // 16, 9, 3
+        System.out.println();
+        System.out.println("Lowest common ancestor: " +
+                lowestCommonAncestor(myTree, myTree.findNode(14), myTree.findNode(5)).getData()); //9
+    }
+
+    // The function receives a Binary Tree and a key
+    // Prints all the ancestors of the key from the root
+    public static <T> void printAncestors(BinaryTree<T> tree, T key) {

```

Comment:
List of ancestors of a key seems like a potentially useful thing. What if in the future we want to do something interesting with it rather than just print it?
Could you please refactor code to separate finding ancestors from printing them?

Also printAncestors operates on internal data of BinaryTree class. It would be nice to keep operations that touch internal data within the class that contains this data. Could you move methods that operate on BinaryTree to BinaryTree class?